tests/ModelContextProtocol.Tests/Configuration/McpServerBuilderExtensionsPromptsTests.cs

```diff
@@ -282,4 +283,8 @@ public class ObjectWithId
     {
         public string Id { get; set; } = Guid.NewGuid().ToString("N");
     }
+
+    [JsonSerializable(typeof(ObjectWithId))]
+    [JsonSerializable(typeof(PromptMessage))]
+    partial class JsonContext4 : JsonSerializerContext;
```

Comment:
It'd be nice to address the source generator limitation around such nested types requiring unique names. Do we have an issue tracking that in runtime?

Reply:
Yep https://github.com/dotnet/runtime/issues/72671